src/core/task/return.cc

```diff
@@ -30,6 +33,111 @@ using namespace Legion;
 
 namespace legate {
 
+struct JoinReturnedException {
+  using LHS = ReturnedException;
+  using RHS = LHS;
+
+  static const ReturnedException identity;
+
+  template <bool EXCLUSIVE>
+  static void apply(LHS& lhs, RHS rhs)
+  {
+    assert(EXCLUSIVE);
```

Comment:
Should this be guarded by `DEBUG_LEGATE`? Or possibly have two specializations, for `EXCLUSIVE=true` and `EXCLUSIVE=false`, where the latter just does `assert(false)`?

Reply:
7feb1761de1c7dc4c1fc2c73f682cb3682590794